mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/query/CriterionConverter.java

```diff
@@ -87,26 +88,41 @@ private void registerCriterionConverters() {
     private <T extends Criterion> void registerCriterionConverter(Class<T> type, Function<T, QueryBuilder> f) {
         criterionConverterMap.put(type, (Function<Criterion, QueryBuilder>) f);
     }
-    
+
+    private QueryBuilder createHeaderNameQuery(String headerName) {
```

Comment:
You need to still handle the logic of querying old data: we did not migrate yet!

Please use a OR query with the old one...